.chronus/changes/change-some-bad-apis-2024-8-3-7-4-12.md

```diff
@@ -0,0 +1,12 @@
+---
+# Change versionKind to one of: internal, fix, dependencies, feature, deprecation, breaking
+changeKind: breaking
+packages:
+  - "@azure-tools/typespec-client-generator-core"
+---
+
+1. The kind for `unknown` renamed from `any` to `unknown`.
+2. The `values` property in `SdkUnionType` renamed to `variantTypes`.
```

Comment:
`values` is different than `variantTypes` though, shouldn't it correspond to `variants`? Same for tuples

Reply:
I would like to have the `types` suffix because in other places we have `types` such as in `SdkArrayType` we have `valueType` - therefore I think maybe we should keep consistent

Reply:
oh - and it is the type of the variant or in this way: in this context, the variant of the union is a type

Reply:
I had an offline discussion with @tadelesh 
so our current naming convention is adding the `type` suffix everywhere, it would be weird if we do not do the same here, for example:
- the type names we have type suffix: `SdkModelType`, `SdkEnumType`, etc
- the properties inside a type if it is referring another type, we have the type suffix, such as the `keyType` `valueType` in `SdkDictionaryType`, `wireType` and `baseType` in `SdkDateTimeType`

So if we do not want the type suffix, we should do it consistently and thoroughly, we might have to remove all of them, which keep consistent with typespec apis: for type names we are adding a `Sdk` as prefix, such as `SdkModel`, `SdkClient`, `SdkEnum`, and those properties we do not need the `type` suffix such as we could have `wire` and `base` in `SdkDateTime`.

I think we could think this through, and if we decide to make this alignment, we could do it in next month's release

Reply:
I thought these `variantTypes` were referring to the `variant` values themselves, not the types of the variants. If it's the types of the `variant`s then `variantTypes` is appropriate. I'm not sure we need a total redesign if that's the case, do you agree?

Reply:
maybe we could put that as a future item to discuss.
At least now our types are consistent with each other on their names.
But whether we want to align with the style in typespec compiler (because it would be a big change), we could discuss that later.